packages/protocol/src/action/server/FilteredAttributeResponse.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * @license
+ * Copyright 2022-2025 Project CHIP Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { AttributeTypeProtocol, ClusterProtocol, EndpointProtocol, NodeProtocol } from "#action/protocols.js";
+import { ReadResult } from "#action/response/ReadResult.js";
+import { InternalError } from "#general";
+import { AttributeId, AttributePath, ClusterId, EndpointNumber } from "#types";
+import { AccessControl } from "./AccessControl.js";
+import { AttributeResponse } from "./AttributeResponse.js";
+
+type ClusterFilter = {
+    [clusterId: ClusterId]: Set<AttributeId>;
+};
+export type AttributeResponseFilter = {
+    [endpointId: EndpointNumber]: ClusterFilter;
+};
+
+/**
+ * FilteredAttributeResponse is a specialized version of AttributeResponse that processes a read/subscribe request
+ * with a filter applied to the attributes. Only the attributes that match the filter will be processed.
+ */
+export class FilteredAttributeResponse<
```

Comment:
Maybe call it "AttributeSubscriptionResponse" since AttributeResponse is also filtered?

Reply:
good idea